From: Chris Coulson <chris.coulson@canonical.com>
Date: Thu, 15 Mar 2018 23:54:34 +0000
Subject: Handle the case where passwd re-asks for the current password when
 it was entered incorrectly (when using LDAP)

Based on a patch from Ryan Tandy
Bug-Ubuntu: https:/launchpad.net/bugs/607357
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=786530
Forwarded: yes
---
 panels/user-accounts/run-passwd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: gnome-control-center-3.32.1/panels/user-accounts/run-passwd.c
===================================================================
--- gnome-control-center-3.32.1.orig/panels/user-accounts/run-passwd.c
+++ gnome-control-center-3.32.1/panels/user-accounts/run-passwd.c
@@ -404,7 +404,7 @@ io_watch_stdout (GIOChannel *source, GIO
 
                         if (is_string_complete (str->str, "assword: ", "failure", "wrong", "error", NULL)) {
 
-                                if (strstr (str->str, "assword: ") != NULL) {
+                                if (strstr (str->str, "assword: ") != NULL && strstr (str->str, "incorrect") == NULL) {
                                         /* Authentication successful */
 
                                         passwd_handler->backend_state = PASSWD_STATE_NEW;
